fe/fe-core/src/main/java/com/starrocks/sql/optimizer/RequiredPropertyDeriver.java

```diff
@@ -258,4 +263,48 @@ public Void visitPhysicalNoCTE(PhysicalNoCTEOperator node, ExpressionContext con
         requiredProperties.add(Lists.newArrayList(requirementsFromParent));
         return null;
     }
+
+    private class CTEPropertyDeriver extends OperatorVisitor<Void, Void> {
+        @Override
+        public Void visitOperator(Operator node, Void context) {
+            if (requirementsFromParent.getCteProperty().isEmpty()) {
+                return null;
+            }
+
+            // Pass CTE property to children
+            for (List<PhysicalPropertySet> requiredProperty : requiredProperties) {
+                for (int i = 0; i < requiredProperty.size(); i++) {
+                    PhysicalPropertySet property = requiredProperty.get(i).copy();
+                    property.setCteProperty(requirementsFromParent.getCteProperty());
+                    requiredProperty.set(i, property);
+                }
+            }
+
+            return null;
+        }
+
+        @Override
+        public Void visitPhysicalCTEAnchor(PhysicalCTEAnchorOperator node, Void context) {
+            visitOperator(node, context);
+            PhysicalPropertySet requiredRight = requiredProperties.get(0).get(1);
+
+            CTEProperty thisCTE = new CTEProperty(node.getCteId());
+            thisCTE.merge(requiredRight.getCteProperty());
+            PhysicalPropertySet copy = requiredRight.copy();
+
+            copy.setCteProperty(thisCTE);
+            requiredProperties.get(0).set(1, copy);
```

Comment:
If you set the copy property back, why do we need the copy?

Reply:
Don't modify origin property(maybe global property: EMPTY)